mmdet/datasets/pipelines/transforms.py

```diff
@@ -373,35 +382,43 @@ def __call__(self, results):
             results[key] = img
         results['img_shape'] = img_shape
 
+        valid_flag = False
         # crop bboxes accordingly and clip to the image boundary
         for key in results.get('bbox_fields', []):
+            # e.g. gt_bboxes and gt_bboxes_ignore
             bbox_offset = np.array([offset_w, offset_h, offset_w, offset_h],
                                    dtype=np.float32)
             bboxes = results[key] - bbox_offset
             bboxes[:, 0::2] = np.clip(bboxes[:, 0::2], 0, img_shape[1])
             bboxes[:, 1::2] = np.clip(bboxes[:, 1::2], 0, img_shape[0])
-            results[key] = bboxes
+            valid_inds = (bboxes[:, 2] > bboxes[:, 0]) & (
+                bboxes[:, 3] > bboxes[:, 1])
+            # When there is no gt bbox, cropping is conducted.
+            # When the crop is valid, cropping is conducted.
+            if len(valid_inds) == 0 or valid_inds.any():
+                valid_flag = True
+            results[key] = bboxes[valid_inds, :]
+            # label fields. e.g. gt_labels and gt_labels_ignore
+            label_key = key.replace('bboxes', 'labels')
```

Comment:
Some keys may not contain the substring `bboxes`, such as `proposals`. In this case `label_key` is just `key` and the next two lines will result in errors.

Reply:
Keen eyes! I have added bbox2label and bbox2mask key correspondence in RandomCrop and MinIouRandomCrop classes. Please have a look.